### PR TITLE
[WIP][Harness] Move all common code to the shared lib.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Moved all the code that can be shared with the CLI to the common
library. We neede some small changes, but mainly due to namespaces and a
forgotten interface implemenation (was already implemented, was missing
in the class).

This was fun :)

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Created as a draft to simplify the review. Mostly is files being moved.

**Comment by mandel-macaque** [comment]
This is a draft for real, @premun and I have to work on it a little to clean things up

File: tests/xharness/xharness.csproj [hunk 1]
```diff
@@ -73,11 +73,7 @@
     <PackageReference Include="Mono.Options" Version="6.6.0.161" />
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="AppRunner.cs" />
```
Comment (premun) [inline tests/xharness/xharness.csproj:76]: From these, we will need `CrashSnapshotReporterFactory`, `DeviceLogCapturerFactory` and things you will need for app bundling.

We won't really need Simulator/DeviceLoader factories and you will see about RunTasks as per our conversation offline

File: tests/xharness/Harness.cs [hunk 2]
```diff
@@ -12,14 +12,6 @@
 using Xharness.Targets;
 
 namespace Xharness {
-	public enum HarnessAction {
```
Comment (premun) [inline tests/xharness/Harness.cs:15]: Everything Harness and AppRunner should stay behind

File: tests/xharness/Targets/Target.cs [hunk 3]
```diff
@@ -4,8 +4,9 @@
 using System.Linq;
 using System.Xml;
 using Microsoft.DotNet.XHarness.iOS.Shared.Utilities;
+using Microsoft.DotNet.XHarness.iOS.Shared;
 
-namespace Xharness.Targets
+	namespace Xharness.Targets
```
Comment (rolfbjarne) [inline tests/xharness/Targets/Target.cs:9]:
```suggestion
namespace Xharness.Targets
```

**Pull request merged**